src/audio/module_adapter/module_adapter.c

```diff
@@ -240,9 +241,17 @@ static int module_adapter_dp_queue_prepare(struct comp_dev *dev)
 			 &sink_buffer->stream.runtime_stream_params,
 			 sizeof(sink_buffer->stream.runtime_stream_params));
 		/* calculate time required the module to provide OBS data portion - a period */
-		unsigned int sink_period = 1000000 * sink_get_min_free_space(mod->sinks[i]) /
+		uint32_t min_free_size =
+			sink_get_frame_bytes(mod->sinks[i]) *
+			(sink_get_rate(mod->sinks[i]) / LL_TIMER_PERIOD_US);
+		min_free_size = MIN(min_free_size, sink_get_min_free_space(mod->sinks[i]));
+
+		unsigned int sink_period = 1000000 * min_free_size /
 					   (sink_get_frame_bytes(mod->sinks[i]) *
 					   sink_get_rate(mod->sinks[i]));
+		/* Round down period to integer number of LL ticks */
+		sink_period = (sink_period / LL_TIMER_PERIOD_US) * LL_TIMER_PERIOD_US;
+
```

Comment:
you may not need so much complicated calculation, please check:
https://github.com/thesofproject/sof/pull/8412/commits/aa04f1d385c5f778bb2cacb3ad759fb1fb6c7abd
src obs are simplified, this is due to actual buffer size = obs * 2, so do not need left 5 words redundancy.

Reply:
You are right, but there are configurations that sets OBS to value which gives period as a fraction of LL timer period (eg. from reference firmware). To keep compatibility I added this round down calculation.